### PR TITLE
redis proxy : read commands policy for prefix route

### PR DESCRIPTION
**Pull request by ikenchina**
It is useful for the ability to route read commands to another cluster in some cases.
e.g. Mirror writes commands to another Redis cluster in a remote datacenter but reads the Redis cluster in the local datacenter.

Commit Message: read commands policy for prefix route
Additional Description:
Risk Level: Low
Testing: Ut
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto [hunk 1]
```diff
@@ -118,6 +118,7 @@ message RedisProxy {
   }
 
   message PrefixRoutes {
+    // [#next-free-field: 6]
```
Comment (adisuissa) [inline api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto:121]: v2 of the API is frozen (immutable). Please revert the changes to this file and only apply the changes to the v3 proto.

File: api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto [hunk 2]
```diff
@@ -168,6 +168,10 @@ message RedisProxy {
         bool exclude_read_commands = 3;
       }
 
+      message ReadCommandPolicy {
+        string cluster = 1 [(validate.rules).string = {min_len: 1}];
```
Comment (adisuissa) [inline api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto:172]: Please add comments describing the new fields.

**Review by adisuissa — COMMENTED** [review]
Thanks!
Left a few comments on the API changes.
@weisisea can you PTAL at this PR as code-owner?

File: api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto (hunk 1 above)
Comment (ikenchina) [reply api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto]: fixed

File: api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto (hunk 2 above)
Comment (ikenchina) [reply api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto:173]: fixed

File: api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto [hunk 3]
```diff
@@ -168,6 +168,11 @@ message RedisProxy {
         bool exclude_read_commands = 3;
       }
 
+      // ReadCommandPolicy controls how Envoy routes read commands to another cluster.
```
Comment (mattklein123) [inline api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto:171]:
```suggestion
      // ReadCommandPolicy specified that Envoy should route read commands to another cluster.
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM. Please add a release note, thanks.

/wait

**Review by mattklein123 — APPROVED** [review]
Thanks!
